logs/file.go

```diff
@@ -210,50 +236,47 @@ func (w *fileLogWriter) lines() (int, error) {
 
 // DoRotate means it need to write file in new file.
 // new file name like xx.2013-01-01.2.log
-func (w *fileLogWriter) DoRotate() error {
+func (w *fileLogWriter) doRotate() error {
 	_, err := os.Lstat(w.Filename)
-	if err == nil { // file exists
-		// Find the next available number
-		num := 1
-		fname := ""
-		suffix := filepath.Ext(w.Filename)
-		filenameOnly := strings.TrimSuffix(w.Filename, suffix)
-		if suffix == "" {
-			suffix = ".log"
-		}
-		for ; err == nil && num <= 999; num++ {
-			fname = filenameOnly + fmt.Sprintf(".%s.%03d%s", time.Now().Format("2006-01-02"), num, suffix)
-			_, err = os.Lstat(fname)
-		}
-		// return error if the last file checked still existed
-		if err == nil {
-			return fmt.Errorf("Rotate: Cannot find free log number to rename %s\n", w.Filename)
-		}
-
-		// block Logger's io.Writer
-		w.mw.Lock()
```

Comment:
no need to lock here,we use a double-lock to guarantee this function be run thread-safe